src/renderer/reader/components/ReaderMenu.tsx

```diff
@@ -581,6 +578,11 @@ const AnnotationList: React.FC<{ r2Publication: R2Publication, dockedMode: boole
     };
 
     const [pageNumber, setPageNumber] = React.useState(getStartPage);
+    React.useEffect(() => {
+        if (pageNumber > pageTotal) {
+            setPageNumber(pageTotal);
+        }
+    }, [pageTotal]);
```

Comment:
dependency array, missing `pageNumber`?

Reply:
I don't think so, I want to trigger the test on pageTotal change

Reply:
doesn't this break rules of hooks?

> The array of dependencies is not passed as arguments to the effect function. Conceptually, though, that’s what they represent: every value referenced inside the effect function should also appear in the dependencies array. In the future, a sufficiently advanced compiler could create this array automatically

https://legacy.reactjs.org/docs/hooks-reference.html

Reply:
> make sure the array includes all values from the component scope (such as props and state) that change over time and that are used by the effect. Otherwise, your code will reference stale values from previous renders.

https://legacy.reactjs.org/docs/hooks-reference.html

Reply:
`pageTotal` n'est ni un état ni un props 🙃

Reply:
any value within the closure scope (i.e. render function, given that hooks are always top-level) that participates to computations inside the hook's closure should be declared explicitly as a dependency (in fact a linter would probably yell at us for not following this rule).
but I don't fully understand what the logic is here so you probably have something in mind that justifies firing the effect for only a subset of conputation dependencies?

Reply:
ok, let's merge this PR, I trust you judgement regarding the runtime logic.